packages/vuetify/src/mixins/intersectable/index.ts

```diff
@@ -0,0 +1,43 @@
+import Vue from 'vue'
+import { consoleWarn } from '../../util/console'
+
+export default function intersectable (options: { onVisible: string[] }) {
+  if (!('IntersectionObserver' in window)) {
+    return Vue.extend({
+      name: 'intersectable',
+      // do nothing because intersection observer is not available
+    })
+  }
+
+  return Vue.extend({
+    name: 'intersectable',
+
+    mounted () {
+      const observer = new IntersectionObserver((
+        entries: IntersectionObserverEntry[] = [],
+      ) => {
+        this.onObserve(entries)
+      })
+
+      observer.observe(this.$el)
+    },
+    methods: {
+      onObserve (entries: IntersectionObserverEntry[]) {
+        const isVisible = Boolean(entries.find(entry => entry.isIntersecting))
+
+        if (!isVisible) {
+          return
+        }
+
+        for (let i = 0, length = options.onVisible.length; i < length; i++) {
+          const callback = (this as any)[options.onVisible[i]]
+          if (typeof callback === 'function') {
```

Comment:
opt for early return in this conditional.

```
if (typeof callback === 'function') {
  callback()

  return
}
```

Reply:
But we need to call all callbacks, not just first

Reply:
sorry, `continue`